spring-cloud-stream-binder-kafka-core/src/main/java/org/springframework/cloud/stream/binder/kafka/provisioning/KafkaTopicProvisioner.java

```diff
@@ -376,32 +376,7 @@ private void createTopicAndPartitions(AdminClient adminClient, final String topi
 		if (names.contains(topicName)) {
 			//check if topic.properties are different from Topic Configuration in Kafka
 			if (this.configurationProperties.isAutoCreateTopics()) {
```

Comment:
I think this should be under a new property `autoAlterTopics` - false by default.

It will still break for users on old brokers where the topics have been changed without reflecting that change in the yaml; previously this would just be ignored.

Reply:
don't mind to add in the current PR, it was my initial plan, but just to make this PR simple kept with regular check

Reply:
@tdanylchuk Please go ahead and add the property in this PR itself if you don't mind.

Reply:
done, please review